builtin/mainmenu/common.lua

```diff
@@ -171,6 +171,23 @@ os.tempfolder = function()
 	local filetocheck = os.tmpname()
 	os.remove(filetocheck)
 
+	-- https://blogs.msdn.microsoft.com/vcblog/2014/06/18/c-runtime-crt-features-fixes-and-breaking-changes-in-visual-studio-14-ctp1/
+	--   The C runtime (CRT) function called by os.tmpname is tmpnam.
+	--   Microsofts tmpnam implementation in older CRT / MSVC releases is defective.
+	--   tmpnam return values starting with a backslash characterize this behavior.
+	-- https://sourceforge.net/p/mingw-w64/bugs/555/
+	--   MinGW tmpnam implementation is forwarded to the CRT directly.
+	-- https://sourceforge.net/p/mingw-w64/discussion/723797/thread/55520785/
+	--   MinGW links to an older CRT release (msvcrt.dll).
+	--   Due to legal concerns MinGW will never use a newer CRT.
+	--
+	--   Make use of TEMP to compose the temporary filename if an old
+	--   style tmpnam return value is detected.
+	if filetocheck:sub(1, 1) == "\\" then
+		local tempfolder = os.getenv("TEMP")
+		return tempfolder .. filetocheck
```

Comment:
previously `tmpnam` output was only used to determine where to place folder
with this, the (very short) name tmpnam gives it directly

not sure if this is a good idea? is there a reason for this?

Reply:
I believe you are referring to not making use of randname ("MTTempModFolder_XXXXX"), unlike what the 'normal' codepath does?

The code that introduced os.tempfolder before https://github.com/minetest/minetest/commit/a2de43, dating to somewhere in 2014 https://github.com/minetest/minetest/commit/c39845
, uses this same short-name construct:

```
if DIR_DELIM == "\\" then
    local tempfolder = os.getenv("TEMP")
    return tempfolder .. filetocheck
```

The reason to keep this code is because it seems to have survived in the codebase for years.
Is redesign wanted?

Reply:
> uses this same short-name construct:

oh indeed it does, just remembered it incorrectly